packages/protocol/test/authorizers/Api3TokenLock.sol.js

```diff
@@ -0,0 +1,702 @@
+
+        // set api3 authorizer
+        await api3TokenLock.connect(admin).setApi3RequesterRrpAuthorizer(api3RequesterRrpAuthorizer.address);
+
+        // set api3 token
+        await api3TokenLock.connect(admin).setApi3Token(api3Token.address);
+
+        // approve Api3TokenLock:
+        await api3Token.connect(alice).approve(api3TokenLock.address, await api3TokenLock.lockAmount());
+        // Lock:
+        await api3TokenLock.connect(alice).lock(airnodeId, bobClient.address);
+      });
+
+      context('caller is unlocking successfully', async () => {
+        beforeEach(async () => {
+          timeTravel(ONE_MINUTE_IN_SECONDS + 1000);
+        });
+
+        it('is set', async () => {
+          const beforeBalance = await api3Token.balanceOf(alice.address);
+          // when:
+          await api3TokenLock.connect(alice).unlock(airnodeId, bobClient.address);
+          // then:
+          const afterBalance = await api3Token.balanceOf(alice.address);
+          expect(afterBalance).to.equal(beforeBalance.add(LOCK_AMOUNT));
+          expect(await api3TokenLock.canUnlockAt(airnodeId, bobClient.address, alice.address)).to.equal(0);
+          expect(await api3TokenLock.lockAmountAt(airnodeId, bobClient.address, alice.address)).to.equal(0);
+          expect(await api3TokenLock.tokenLocks(airnodeId, bobClient.address)).to.equal(0);
+        });
+
+        it('emits event', async () => {
+          // then:
+          await expect(api3TokenLock.connect(alice).unlock(airnodeId, bobClient.address))
+            .to.emit(api3TokenLock, 'Unlock')
+            .withArgs(airnodeId, bobClient.address, alice.address, LOCK_AMOUNT);
+        });
+      });
+
+      context('caller period not expired', async () => {
+        it('reverts', async () => {
+          // then:
+          await expect(api3TokenLock.connect(alice).unlock(airnodeId, bobClient.address)).to.revertedWith(
+            Errors.LockPeriodNotExpired
+          );
+        });
+      });
+
+      context('already unlocked', async () => {
+        it('reverts', async () => {
+          // given:
+          timeTravel(ONE_MINUTE_IN_SECONDS + 1000);
+          await api3TokenLock.connect(alice).unlock(airnodeId, bobClient.address);
+          // then:
+          await expect(api3TokenLock.connect(alice).unlock(airnodeId, bobClient.address)).to.revertedWith(
+            'No amount locked'
+          );
+        });
+      });
+
+      context('client address is blacklisted', async () => {
+        it('reverts', async () => {
+          // given:
+          await api3TokenLock.setBlacklistStatus(bobClient.address, true);
+          timeTravel(ONE_MINUTE_IN_SECONDS + 1000);
+          // then:
+          await expect(api3TokenLock.connect(alice).unlock(airnodeId, bobClient.address)).to.revertedWith(
+            Errors.ClientBlacklisted
+          );
+        });
+      });
+    });
+
+    describe('withdraw', async () => {
```

Comment:
Same here. Maybe we could try adding more token locks and make sure everything works as expected with more than one.

Reply:
yup done